flujo/domain/dsl/pipeline_step_validations.py

```
@@ -269,18 +290,102 @@ def _validate_pipeline(
+                            else:
+                                merged_paths.update(bpaths)
+                    else:
+                        for bpaths in branch_outputs_by_name.values():
+                            merged_paths.update(bpaths)
+                elif normalized_ms == "overwrite":
+                    # Conservative: only propagate keys we can prove are copied back.
+                    # With context_include_keys this is the included subset; otherwise treat as opaque.
+                    if include_roots and branch_outputs_by_name:
+                        try:
+                            last = sorted(branch_outputs_by_name.keys())[-1]
+                        except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)